### DISCUSSION TITLE
(likely downstream bug, advice requested) diagram result animation not visible in nix build

### DISCUSSION THREAD

**Discussion by ppenguin** [OpenModelica/OpenModelica#10225] [Q&A]
This is likely not an OM bug, but maybe somebody has an idea how/where to look:

I'm running OM on `nixos` directly from `nixpkgs` and it seems to work well, except for diagram animations not being visible.

E.g. when I simulate a `StateGraph` example and play the animation of the diagram, the natively installed `OMEdit` just shows a static diagram, where the same one run in the VM shows the state transitions as expected.

Does anybody have an idea how to debug this issue or whether this might be a silent build error[^1] or could be related to the environment?

Some extra context:
- The `nixpkgs` version I ran is `1.18` and the VM is `1.20`, though I suspect a `1.18` VM would also work correctly
- I ran the native `OMEdit` both under `wayland` and `xwayland`, but not (yet) on "pure" `xorg`. Could this be related as well?

[^1]: At first 3D animation also didn't work in the `nix` package, but this I solved by fixing the package to include the missing `openscenegraph` dependency. This however was also a silent "error", i.e. it installed fine but was missing 3D functionality

**Comment by adeas31** [comment]
Try with 1.18 since a lot of things are fixed.

@anotheruserofgithub any hint?

**Reply by ppenguin** [reply]
Thanks, I tried to build `1.20` in `nixpkgs` but it fails while resolving a library, even though the library is built. This is most likely a `nix` specific issue, I'll be looking into it when I find time. So in short: not yet able to easily try `1.20` as "nix-native".

**Comment by anotheruserofgithub** [comment]
Unfortunately I have no idea. As far as I can tell, OMEdit-specific dependencies are Qt and OSG. Maybe a missing Qt library?

@adeas31 Did you mean "try with 1.20"?

@ppenguin In case you know a bit of C++/Qt and are able to compile the code, here are some pointers to try and debug it:
- https://github.com/OpenModelica/OpenModelica/blob/7954d04c1d1c6a1310d6d941503f05c9e8c79ea0/OMEdit/OMEditLIB/MainWindow.cpp#L3871
- https://github.com/OpenModelica/OpenModelica/blob/7954d04c1d1c6a1310d6d941503f05c9e8c79ea0/OMEdit/OMEditLIB/Plotting/PlotWindowContainer.cpp#L479
- https://github.com/OpenModelica/OpenModelica/blob/7954d04c1d1c6a1310d6d941503f05c9e8c79ea0/OMEdit/OMEditLIB/Plotting/VariablesWidget.cpp#L2577-L2582
- https://github.com/OpenModelica/OpenModelica/blob/7954d04c1d1c6a1310d6d941503f05c9e8c79ea0/OMEdit/OMEditLIB/Plotting/DiagramWindow.cpp#L70

**Reply by ppenguin** [reply]
Thanks, that's might be at least the "hot sections", I'll take a look when I find time. So I guess I'd typically start with adding some debug `printf`'s there to see what's going on?

I guess this is a bit more involved on the `nix` side though (to isolate the deps as a build env and then build OM itself manually) but that'll go some way to try and update to `1.20` as well...

**Comment by anotheruserofgithub** [comment]
@ppenguin This issue was actually solved by https://github.com/OpenModelica/OpenModelica/pull/7975 (commit https://github.com/OpenModelica/OpenModelica/commit/4449cc07e0ae4e1ae06965e42805f01eea0e13fc) but the fix is not available in the [1.18.0](https://github.com/OpenModelica/OpenModelica/releases/tag/v1.18.0) release and was not backported to [1.18.1](https://github.com/OpenModelica/OpenModelica/releases/tag/v1.18.1).

So you should try to install the latest release [1.21.0](https://github.com/OpenModelica/OpenModelica/releases/tag/v1.21.0), and if it doesn't work either like for [1.20.0](https://github.com/OpenModelica/OpenModelica/releases/tag/v1.20.0), you may try [1.19.3](https://github.com/OpenModelica/OpenModelica/releases/tag/v1.19.3) instead.

If still unsuccessful, you should checkout commit https://github.com/OpenModelica/OpenModelica/commit/4449cc07e0ae4e1ae06965e42805f01eea0e13fc (see [this page](https://github.com/OpenModelica/OpenModelica/commits/master?before=5527d073d179f1437f88424d4f963816dc7c2df0+1535&branch=master&qualified_name=refs%2Fheads%2Fmaster) of the history) and compile OM yourself.
I did that as a verification and I can confirm it's working correctly on Ubuntu.
I expect it to work for you as well because the fix was pushed only a few weeks after the 1.18 release, which you are able to run.

@adeas31 I think this discussion can be closed since the reported issue is no longer pending, and I don't think we would be able to provide help for the nix build problems, at least I can't.

**Comment by adeas31** [comment]
> @adeas31 I think this discussion can be closed since the reported issue is no longer pending, and I don't think we would be able to provide help for the nix build problems, at least I can't.

OK. I am closing this one. @ppenguin please reopen if you still encounter issues. We will then try to find resource that can help you.

**Reply by ppenguin** [reply]
@adeas31 @anotheruserofgithub  Thanks, sorry for the silence. Since I needed to get work done with OM I just resorted to improving the docker image and using that, which works nice also on wayland. It has been working ok for me for recent versions (except for the other bugs I reported related to animated diagram output), and it allows me to use the newest one.
